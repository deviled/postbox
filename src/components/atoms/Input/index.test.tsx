import React from 'react';
import { screen, render } from '__tests__/utils';
import userEvent from '@testing-library/user-event';
import { Input } from '.';

describe('Input.tsx', () => {
  const getInput = () => screen.getByRole('textbox');

  it('should match snapshot', () => {
    const { baseElement } = render(<Input type="text" placeholder="Search..." />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should match snapshot with error', () => {
    const { baseElement } = render(<Input type="text" placeholder="Search..." isError />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should have value typed', () => {
    const INPUT_VALUE = 'some input value';

    render(<Input name="test" />);

    userEvent.type(getInput(), INPUT_VALUE);

    expect(getInput()).toHaveValue(INPUT_VALUE);
  });
});
