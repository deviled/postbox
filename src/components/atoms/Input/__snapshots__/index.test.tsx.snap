// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.tsx should match snapshot 1`] = `
.c0 {
  width: auto;
  padding: 8px;
  margin: 4px 0px;
}

<body>
  <div>
    <input
      class="c0"
      placeholder="Search..."
      type="text"
    />
  </div>
</body>
`;
